pkg/ingester/transfer.go

```diff
@@ -275,6 +402,67 @@ func (i *Ingester) transferOut(ctx context.Context) error {
 	return nil
 }
 
+func (i *Ingester) v2TransferOut(ctx context.Context) error {
+	if len(i.dbs) == 0 {
+		level.Info(util.Logger).Log("msg", "nothing to transfer")
+		return nil
+	}
+
+	// Close all user databases
```

Comment:
As for the first comment. The WAL is replayed when the tsdb is opened for a given user, which may not work for large WALs. I've been using 10m blocks in my testing and that particular issue didn't become a problem, though I could see how it might. 

Good call on not closing the dbs. Do you know if it's safe to copy out a tsdb before it's been closed? Maybe the better idea here is to snapshot it and leave it open for queries. Thoughts?